go/git/get.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/keybase/client/go/teams"
+	"github.com/keybase/go-codec/codec"
+)
+
+type ServerResponseRepo struct {
+	TeamID                keybase1.TeamID               `json:"team_id"`
+	RepoID                keybase1.RepoID               `json:"repo_id"`
+	CTime                 time.Time                     `json:"ctime"`
+	MTime                 time.Time                     `json:"mtime"`
+	EncryptedMetadata     string                        `json:"encrypted_metadata"`
+	Nonce                 string                        `json:"nonce"`
+	KeyGeneration         keybase1.PerTeamKeyGeneration `json:"key_generation"`
+	LastModifyingUsername string                        `json:"last_writer_username"`
+	LastModifyingDeviceID keybase1.DeviceID             `json:"last_writer_device_id"`
+}
+
+type ServerResponse struct {
+	Repos []ServerResponseRepo `json:"repos"`
+}
+
+// Implicit teams need to be converted back into the folder that matches their
+// display name. Regular teams become a regular team folder.
+func folderFromTeamID(ctx context.Context, g *libkb.GlobalContext, teamID keybase1.TeamID) (keybase1.Folder, error) {
+	team, err := teams.Load(ctx, g, keybase1.LoadTeamArg{
+		ID: teamID,
+	})
+	if err != nil {
+		return keybase1.Folder{}, err
+	}
+	if team.IsImplicit() {
+		// TODO: This function doesn't currently support conflict info.
+		name, err := team.ImplicitTeamDisplayNameString(ctx)
+		if err != nil {
+			return keybase1.Folder{}, err
+		}
+		var folderType keybase1.FolderType
+		if team.IsPublic() {
+			folderType = keybase1.FolderType_PUBLIC
+		} else {
+			folderType = keybase1.FolderType_PRIVATE
+		}
+		return keybase1.Folder{
+			Name:       name,
+			FolderType: folderType,
+			Private:    !team.IsPublic(),
+		}, nil
+	} else {
+		return keybase1.Folder{
+			Name:       team.Name().String(),
+			FolderType: keybase1.FolderType_TEAM,
+			Private:    !team.IsPublic(),
+		}, nil
+	}
+}
+
+func getMetadataInner(ctx context.Context, g *libkb.GlobalContext, folder *keybase1.Folder) ([]keybase1.GitRepoResult, error) {
+	// TODO: replace these with real implementations
+	var teamer Teamer
+	cryptoer := NewCrypto(g)
+
+	apiArg := libkb.APIArg{
+		Endpoint:    "kbfs/git/team/get",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		NetContext:  ctx,
+		Args:        libkb.HTTPArgs{
+		// a limit parameter exists, default 100, and we don't currently set it
+		},
+	}
+
+	// The team_id parameter is optional. Add it in if the caller supplied it.
+	if folder != nil {
+		if folder.FolderType != keybase1.FolderType_TEAM {
+			return nil, errors.New("git repo folders must be team folders")
+		}
+		teamIDVis, err := teamer.LookupOrCreate(ctx, *folder)
+		if err != nil {
+			return nil, err
+		}
+		apiArg.Args["team_id"] = libkb.S{Val: string(teamIDVis.TeamID)}
+	}
+	res, err := g.GetAPI().Get(apiArg)
+	if err != nil {
+		return nil, err
+	}
+
+	var resObj ServerResponse
+	err = res.Body.UnmarshalAgain(&resObj)
```

Comment:
`GetDecode` is nice: [example](https://github.com/keybase/client/blob/38302635d237216317f281f55ca8d37b2837878a/go/teams/list.go#L39)

Reply:
Fixed.